src/components/contextMenu/AposContextMenu.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div class="apos-context-menu" :class="modifiers">
+    <AposContextMenuTip />
+    <ul class="apos-context-menu__items">
+      <AposContextMenuItem 
+        v-for="item in menu" 
+        :key="item.action" 
+        :menuItem="item" 
+        v-on:clicked="itemClicked"
+      />
+    </ul>
+  </div>
+</template>
+
+<script>
+import AposLoading from '../loading/AposLoading.vue';
+import AposContextMenuItem from './AposContextMenuItem.vue';
+import AposContextMenuTip from './AposContextMenuTip.vue';
+
+export default {
+  name: 'AposContextMenu',
+  components: {
+    AposLoading,
+    AposContextMenuItem,
+    AposContextMenuTip
+  },
+  props: {
+    menu: Array,
+    tipAlignment: {
+      type: String,
+      default: 'left'
+    },
+    origin: {
+      type: String,
+      default: 'below'
+    },
+  },
+  computed: {
+    modifiers() {
+      let modifiers = 
```

Comment:
Default to const

Reply:
Let's do this as an array of classes rather than combining them into a single string. No need to do that work anymore. Could be marginally cleaner by skipping the variable declaration completely.

Reply:
got rid of modifiers at this level because everything gets passed explicitly to tip component